lib/Doctrine/Sniffs/Classes/ExceptionInterfaceNamingSniff.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Sniffs\Classes;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SlevomatCodingStandard\Helpers\UseStatementHelper;
+use Throwable;
+use const T_EXTENDS;
+use const T_INTERFACE;
+use const T_OPEN_CURLY_BRACKET;
+use function array_map;
+use function count;
+use function explode;
+use function in_array;
+use function preg_grep;
+use function strpos;
+
+class ExceptionInterfaceNamingSniff implements Sniff
+{
+    private const CODE_NOT_AN_EXCEPTION = 'NotAnException';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function register()
+    {
+        return [T_INTERFACE];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $importedClassNames = $this->parseImportedClassNames($phpcsFile);
+        $extendedInterfaces = $this->parseExtendedInterfaces($phpcsFile, $stackPtr);
+
+        // Set original classname instead of alias
+        $extendedInterfaces = array_map(function (string $extendedInterface) use ($importedClassNames) {
+            if (isset($importedClassNames[$extendedInterface])) {
+                return $importedClassNames[$extendedInterface];
+            }
+
+            return $extendedInterface;
+        }, $extendedInterfaces);
+
+        $hasExceptionName = strpos($phpcsFile->getDeclarationName($stackPtr), 'Exception') !== false;
+
+        $isExtendingThrowable = (in_array(Throwable::class, $importedClassNames, true) &&
+            in_array(Throwable::class, $extendedInterfaces, true)) ||
+            in_array('\\Throwable', $extendedInterfaces, true);
```

Comment:
No leading `\` should ever be used in const strings (the types are always to be FCQN).
If the `$extendedInterfaces` may contain names with `\`, it needs to be reworked (i.e. `use` resolution moved to `parseExtendedInterfaces()` directly).

Reply:
Should I ignore the leading `\` then because another sniff should handle this smell?

Reply:
The sniff should be able to determine whether `Throwable` is global type or namespaced type (either `use` or `\`), but it shouldn't be hardcoded here like this.